jobsdb/unionQuery.go

```diff
@@ -0,0 +1,284 @@
+package jobsdb
+
+import (
+	"database/sql"
+	"fmt"
+	"strings"
+)
+
+const (
+	selectQuery = `SELECT jobs.job_id, jobs.uuid, jobs.user_id, jobs.parameters, jobs.custom_val, jobs.event_payload, jobs.event_count, jobs.created_at, jobs.expire_at, jobs.customer, sum(jobs.event_count) over (order by jobs.job_id asc) as running_event_counts `
+)
+
+//used to get pickup-counts during server start-up
+func (jd *HandleT) GetCustomerCounts(defaultBatchSize int) map[string]int {
+	unionMap := make(map[string]int)
+	//just using first DS here
+	//get all jobs all over DSs and then calculate
+	//use DSListLock also
+
+	// jd.dsMigrationLock.RLock()
+	jd.dsListLock.RLock()
+	// defer jd.dsMigrationLock.RUnlock()
+	defer jd.dsListLock.RUnlock()
+	rows, err := jd.dbHandle.Query(fmt.Sprintf(`select customer, count(job_id) from %s group by customer;`, jd.getDSList(false)[0].JobTable))
+	jd.assertError(err)
+
+	for rows.Next() {
+		var customer string
+		var count int
+		err := rows.Scan(&customer, &count)
+		jd.assertError(err)
+		unionMap[customer] = count
+	}
+
+	return unionMap
+}
+
+//Unprocessed
+
+func (jd *HandleT) getUnprocessedUnionQuerystring(unionMap map[string]int, ds dataSetT, params GetQueryParamsT) string {
```

Comment:
`unionMap` is not very descriptive, a suggestion:

```suggestion
func (jd *HandleT) getUnprocessedUnionQuerystring(customerCount map[string]int, ds dataSetT, params GetQueryParamsT) string {
```

Reply:
true, it was just a placeholder.
This looks good